kafka-ui-api/src/test/java/com/provectus/kafka/ui/AbstractBaseTest.java

```diff
@@ -1,35 +1,56 @@
 package com.provectus.kafka.ui;
 
-import org.junit.runner.RunWith;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.extension.ExtendWith;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.context.ApplicationContextInitializer;
 import org.springframework.context.ConfigurableApplicationContext;
 import org.springframework.test.context.ActiveProfiles;
-import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
 import org.testcontainers.containers.KafkaContainer;
 import org.testcontainers.containers.Network;
-import org.testcontainers.junit.jupiter.Container;
-import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.containers.wait.strategy.Wait;
 import org.testcontainers.utility.DockerImageName;
 
-@RunWith(SpringRunner.class)
+import java.time.Duration;
+
+@ExtendWith(SpringExtension.class)
 @SpringBootTest
 @ActiveProfiles("test")
-@Testcontainers
 public abstract class AbstractBaseTest {
-    @Container
-    public static KafkaContainer kafka = new KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka:5.2.1"))
-            .withNetwork(Network.SHARED);
-    @Container
-    public static SchemaRegistryContainer schemaRegistry = new SchemaRegistryContainer("5.2.1")
-            .withKafka(kafka)
-            .dependsOn(kafka);
+
```

Comment:
Why did you remove managing test containers lifecycle using `@Container`?

Reply:
Because using `@Container` the containers did restart for every test class, but application properties were not updated with new container addresses

Reply:
Hmm...Ok, I'll check

Reply:
@IldarAlmakaev 

> Containers declared as static fields will be shared between test methods. They will be started only once before any test method is executed and stopped after the last test method has executed. Containers declared as instance fields will be started and stopped for every test method.

I would prefer to do it not in static init but in Initializer initialize method and add shutdown hook

Reply:
and what is the difference between TestContainer's approach and Initializer's one, @germanosin?
As I understand, we would like to create a test container as a singleton for all our tests not only for class level tests.